src/IDCodeValidator.java

```diff
@@ -0,0 +1,175 @@
+            System.out.println("ID is correct");
+            return true;
+        }
+        if (idCode.length() != 11) {
+            System.out.println("The length is not correct");
+
+        }
+
+        return false;
+    }
+
+
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (genderNumber == 1 || genderNumber == 3 || genderNumber ==5){
+            return true;
+        }
+        if (genderNumber == 2 || genderNumber == 4 || genderNumber ==6) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        LocalDateTime birthday = LocalDateTime.of(1998, 01, 29, 0, 0);
+        LocalDateTime now = LocalDateTime.now();
+        long yearsOld = ChronoUnit.YEARS.between(birthday, now);
+        System.out.println(yearsOld);
+
+        if (monthNumber == 2 && isLeapYear(yearNumber) && yearsOld > 0) {
+            System.out.println(yearsOld + 1);
+            return true;
+        }
+        if ((!(yearsOld > 0))) {
+            System.out.println("Error");
+        }
+
+        return false;
+    }
+
+
+    public static boolean isLeapYear(int yearNumber) {
+        if ((yearNumber % 4 == 0 && yearNumber % 100 != 0) || (yearNumber % 400 == 0)) {
+            return true;
+        } else return false;
+    }
+
+
+    public static boolean checkControlNumber(String idCode) {
+        int result = 0;
+        List<Integer> a = new ArrayList<>();
+        a.add(Integer.valueOf(idCode));
+        for (int i = 0; i < idCode.length(); i++) {
+            String n = String.valueOf(idCode.charAt(i));
+            a.add(Integer.parseInt(n));
+        }
+        List<Integer> x = new ArrayList<>();
+        x.add(1);
+        x.add(2);
+        x.add(3);
+        x.add(4);
+        x.add(5);
+        x.add(6);
+        x.add(7);
+        x.add(8);
+        x.add(9);
+        x.add(1);
+        for (int i = 0; i < a.size() - 1; i++) {
+            result = result + (a.get(i) * x.get(i));
+        }
+        int leftover = result % 11;
+
+        if (leftover < 10) {
+            System.out.println(result);
+        }
+        if (leftover == 10) {
+            System.out.println(result == 0);
+        }
+        if (leftover > 10) {
+            System.out.println("Error");
+        }
+        return false;
```

Comment:
This is not really correct.
If the remainder (leftOver) is less than 10, then we check that this number matches with the last number (control number)
If the remainder is equal to 10, then we do another set of calculations. See [here](https://github.com/tallinn-learning-ilja/HW06/blob/0b88b9f608c22b62a0e140461da3a93074f83a06/readme.md?plain=1#L38)
Remainder can never be greater than 10 because then it would divide by 11 and remainder would be actually 0. So you can remove 
```
    if (leftover > 10) {
            System.out.println("Error");
        }
 ```